### PR TITLE
fix: Add slug to articles

### PR DESCRIPTION
**Pull request by tejaswinichile**
Fixes #575

### CODE REVIEW

File: db/migrate/20220926164441_add_slug_to_article.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    change_column_null(:articles, :slug, false)
+  end
+
+  def down
+    remove_column(:articles, :slug)
+  end
+
+  def update_past_articles_with_slug
+    Article.all.each do |article|
+      slug = article.title.underscore.parameterize(separator: '_')
```
Comment (sojan-official) [inline db/migrate/20220926164441_add_slug_to_article.rb:17]: We might have to add a fallback condition here for the cases where someone created articles with the same title. 
We could catch the error and try updating the slug with numerical increments in the end.

File: db/migrate/20220926164441_add_slug_to_article.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+class AddSlugToArticle < ActiveRecord::Migration[6.1]
+  def up
+    add_column :articles, :slug, :string, default: ''
```
Comment (sojan-official) [inline db/migrate/20220926164441_add_slug_to_article.rb:3]: is the default: '' required ?

File: db/migrate/20220926164441_add_slug_to_article.rb (hunk 1 above)
Comment (pranavrajs) [reply db/migrate/20220926164441_add_slug_to_article.rb:17]: The separator should be `-` not `_`

File: db/migrate/20220926164441_add_slug_to_article.rb (hunk 2 above)
Comment (tejaswinichile) [reply db/migrate/20220926164441_add_slug_to_article.rb:3]: When null is `false` we need to have some default value. This usually helps with the previous records which are missed with the slug.

**Review by sojan-official — APPROVED** [review]

**Pull request merged**